subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/cntr.c

```diff
@@ -47,17 +116,89 @@ uint32_t cntr_stop(void)
 		return 1;
 	}
 
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+	/* TODO: if we own and stop GRTC, implement stop here */
+#else /* !CONFIG_BT_CTLR_NRF_GRTC */
 	nrf_rtc_task_trigger(NRF_RTC, NRF_RTC_TASK_STOP);
+#endif /* !CONFIG_BT_CTLR_NRF_GRTC */
 
 	return 0;
 }
 
 uint32_t cntr_cnt_get(void)
 {
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+	uint32_t cntr_l, cntr_h, cntr_h_overflow;
+
+	/* NOTE: For a 32-bit implementation, L value is read after H
+	 *       to avoid another L value after SYSCOUNTER gets ready.
+	 *       If both H and L values are desired, then swap the order and
+	 *       ensure that L value does not change when H value is read.
+	 */
+	do {
+		cntr_h = nrf_grtc_sys_counter_high_get(NRF_GRTC);
+		cntr_l = nrf_grtc_sys_counter_low_get(NRF_GRTC);
+		cntr_h_overflow = nrf_grtc_sys_counter_high_get(NRF_GRTC);
+	} while ((cntr_h & GRTC_SYSCOUNTER_SYSCOUNTERH_BUSY_Msk) ||
```

Comment:
This logic seems incorrect.
You should be reading the low part first and then the high.
If the high does not have the overflow bit set, the whole 52 bit value is correct.

(Reading the high first is against the spec, and I presume you will be very likely to read an overflow from that first high read, as that would be relative to the last time the low was read)

Reply:
`nrfy_grtc_sys_counter_get()`

Reply:
Discussed offline, rewriting this in a more optimal way can be done later